pkg/plugins/placementpolicy/placementpolicy.go

```diff
@@ -404,3 +379,20 @@ func groupPodsBasedOnNodePreference(podList []*corev1.Pod, pod *corev1.Pod, node
 
 	return podsOnNodeWithMatchingLabels
 }
+
+// assignedPod selects pods that are assigned (scheduled and running).
```

Comment:
What if the pod is not assigned yet but in the pre-binding state?

Reply:
It probably makes more sense to act of deleted pods regardless as to whether they had been assigned a node or not. I'll remove this function & update the event handler accordingly.